fixing some bugs

bmcculley opened the pull request:
python 3 had some issues with mixed tabs and spaces. Squashed a couple c bugs (comparison type warnings).
I didn't get it working with python 3 though, I'm pretty sure it would break python 2. I'll have to look at it more.

Commit:
fixing some bugs

python 3 had some issues with mixed tabs and spaces. Squashed a couple c bugs.

jslegendre commented on binddump.c:
```diff
@@ -60,7 +60,7 @@ void free_dyld_cmd(void **dyld_cmd) {
 void free_dyld_array(dyld_t *dylds) {
 	size_t arr_size = sizeof(dylds)/sizeof(dylds[0]);
-	for(int i = 0; i < arr_size; i++) {
+	for(int i = 0; (size_t)i < arr_size; i++) {
```
The thing about C is that all arithmetic types convert implicitly. A compiler may throw a warning up but the conversion is still correct.

That being said, `size_t` is an `unsigned` type so the type of `i` (could/should) be changed to an `unsigned` type.  I have been using `uintptr_t` in this project because it deals with files and `uintptr_t`  is the size of a pointer on a given system.

jslegendre commented on binddump.c:
```diff
@@ -267,13 +267,13 @@ void print_info(dyld_t *dylds, uintptr_t ndyld) {
         if(dylds[i].nlazy_syms) {
             printf("    Lazy:\n");
-            for(int j = 0; j<dylds[i].nlazy_syms; j++) {
+            for(int j = 0; (unsigned)j<dylds[i].nlazy_syms; j++) {
```
This comment is for both this change and the change on line 270.

Casting (in general) should only be done when there is no way around it.  Especially casting between `signed` (i.e. int) and `unsigned` because it can cause some unexpected behavior.  The best thing to do here would be to change the initial type of `j` to an `unsigned` type instead of casting it later on as there is no need for `j` to ever be `signed` in this case.

jslegendre reviewed (changes requested):
Great job picking up on those C inconsistencies! I know learning all of the "rules" when it comes to these lower level languages can be challenging at first.  Just make a couple of those changes and we should be good to go on the merge 👍

jslegendre commented:
Also, fantastic job on the Python stuff.  I never would have thought to go back and format the tabs/spaces.  Also, it was written for Python 2 as that is what comes pre-installed on macOS.

Commit:
changes

switch from casting to uintptr_t which is a better way to handle the compiler warnings.

bmcculley commented:
Sorry, finally made the changes.

jslegendre commented:
Good work man!
